src/pgrn-custom-scan.c

```diff
@@ -203,8 +252,26 @@ PGrnExplainCustomScan(CustomScanState *node, List *ancestors, ExplainState *es)
 }
 
 static void
-PGrnEndCustomScan(CustomScanState *node)
+PGrnEndCustomScan(CustomScanState *customScanState)
 {
+	PGrnScanState *state = (PGrnScanState *) customScanState;
+	unsigned int nTargetColumns;
+
+	ExecClearTuple(customScanState->ss.ps.ps_ResultTupleSlot);
+
+	if (state->tableCursor)
+	{
+		grn_table_cursor_close(ctx, state->tableCursor);
+		state->tableCursor = NULL;
+	}
+	nTargetColumns = GRN_PTR_VECTOR_SIZE(&(state->columns));
+	for (unsigned int i = 0; i < nTargetColumns; i++)
+	{
+		grn_obj *column = GRN_PTR_VALUE_AT(&(state->columns), i);
+		grn_obj_unlink(ctx, column);
+	}
+	GRN_OBJ_FIN(ctx, &(state->columns));
+	GRN_OBJ_FIN(ctx, &(state->columnValue));
```

Comment:
It would be better to free in order as below.
Because we allocated `state->columnValue` after `state->columns`.


```
GRN_OBJ_FIN(ctx, &(state->columnValue));
GRN_OBJ_FIN(ctx, &(state->columns));
```